docker/build.sh

```diff
@@ -1,4 +1,10 @@
-#!/bin/bash
+
+# Check if associative arrays arrays are supported
+unset assoc
+if ! declare -A assoc > /dev/null 2>&1 ; then
```

Comment:
A comment from @gsiv got lost I think:

How about we actually check the Bash version?  Example:

```bash
[[ "${BASH_VERSINFO[0]:-0}" -ge 4 ]] || {
  echo "Ancient bash version :("
  exit 4
}
```

Finally, please don't remove the shebang.

Reply:
https://github.com/OpenSlides/OpenSlides/commit/608fcb38e4ff9680fc3c57f2e12dde5f1aede849

Reply:
Yes, I had considered it but decided to go the other way. Also I think my suggestion clearly states what the problem is. I would not change it.

Reply:
> Yes, I had considered it but decided to go the other way.

Would you like to share your considerations?  I'm curious.

If we were to opt for the proposed method, I would like to point out that it does not make sense to a) `unset` a variable that does not exist and b) create an array that we won't need.  Instead of declaring `assoc`, I would suggest to simply catch the error when we declare `TARGETS`.

Reply:
I didn't overthink it. I was just looking for something that resolves the issue, not the perfect solution. For me it seemed better to check `assoc` instead of the bash version. And I took a decision. I don't care if we do it the other way and your suggestion sounds good, so I implemented it. Let's merge this and move forward.